ui/src/app/join-lobby-link/join-lobby-link.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+import { ApiService } from '../_services/api.service';
+import { ToastrService } from 'ngx-toastr';
+import { RedirectionService } from '../_services/redirection.service';
+import { ActivatedRoute } from '@angular/router';
+import { FormBuilder, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-join-lobby-link',
+  templateUrl: './join-lobby-link.component.html',
+  styleUrls: ['./join-lobby-link.component.scss']
+})
+export class JoinLobbyLinkComponent implements OnInit {
+
+  lobbyId;
+
+  joinLobbyForm = this.fb.group({ userName: ['', Validators.required ],});
+
+  constructor(
+    private api: ApiService,
+    private toaster: ToastrService,
+    private redirectionService: RedirectionService,
+    private route: ActivatedRoute,
+    private fb: FormBuilder,
+  ) { }
+
+  ngOnInit() {
+  }
+
+  onNoClick(): void {
+    this.redirectionService.redirectToHomePage();
+  }
+
+  onJoinClick(): void {
+    this.route.paramMap.subscribe(params => {
+        this.lobbyId = params.get('lobbyId');
+      });
+    if (this.joinLobbyForm.valid) {
+      const userName = this.joinLobbyForm.controls['userName'].value;
+      this.api.joinLobby(this.lobbyId, userName).subscribe(
+        () => {
+          this.redirectionService.redirectToSingleLobby(this.lobbyId);
+        },
+        error => {
+          this.toaster.error(error);
+        },
+      ); 
+    }
+  }
```

Comment:
You don't need the lobby id from if `joinLobbyForm` is not valid, so you can move it inside the if.

Also it would be much cleaner to execute a chain of operations here using methods from rxjs library.
Look at the solution here: https://privatebin.net/?27d08c0a19bd0951#FY5xmvH14xn8eN69bWC8My7j8pAiXagYdUuWPuLypdDj

What's happening here?
1. We get the `lobbyId` from url
    if we succeed then
2. We pass this `lobbyId` to the next Observable and execute it.
3. Of course to make any of above steps happen we need to `subscribe()` to our newly created chain of streams.

More about functions that I used:
`pipe`:  https://www.learnrxjs.io/learn-rxjs/concepts/rxjs-primer#pipe
`pluck`: https://www.learnrxjs.io/learn-rxjs/operators/transformation/pluck
`switchMap`: https://www.learnrxjs.io/learn-rxjs/operators/transformation/switchmap